packages/notifi-react-card/lib/components/subscription/EventTypeHealthCheckRow.tsx

```diff
@@ -53,52 +59,83 @@ export const EventTypeHealthCheckRow: React.FC<
     ratios = config.checkRatios.value;
   }
 
-  const enabled = !loading && !!alerts[alertName];
-
   useEffect(() => {
-    if (isValueType) {
-      setSelectedRatio(ratios[-1]?.ratio);
+    if (loading) {
+      return;
+    }
+    const alert = alerts[alertName];
+    const checkRatios = ratios.map((ratio) => ratio.ratio);
+    if (alert) {
+      const alertRatioValue = JSON.parse(alert.filterOptions ?? '').threshold;
+      setEnabled(true);
+      ratios.forEach((ratio, index) => {
+        if (ratio.ratio === alertRatioValue && customValue === '') {
+          setSelectedIndex(index);
+        }
+      });
+      setInitialRatio(alertRatioValue);
+      if (!checkRatios.includes(alertRatioValue) && customValue === '') {
+        setSelectedIndex(3);
+        setCustomValue(alertRatioValue * 100 + '%');
+      }
+    } else {
+      setEnabled(false);
       setSelectedIndex(ratios.length - 1);
+      setInitialRatio(ratios[ratios.length - 1]?.ratio);
     }
-  }, [isValueType]);
+  }, [
+    alertName,
+    alerts,
+    loading,
+    ratios,
+    setEnabled,
+    setCustomValue,
+    setSelectedIndex,
+  ]);
 
-  const handleToggleNewSubscription = () => {
+  const handleToggleNewSubscription = useCallback(() => {
     if (loading) {
       return;
     }
-    setIsShowSelectButton(!isShowSelectButton);
-    // TODO: hook up API call here
-    // if (!enabled) {
-    //   instantSubscribe({
-    //     alertConfiguration: healthThresholdConfiguration({
-    //       alertFrequency: 'ALWAYS',
-    //       percentage: selectedRatio,
-    //     }),
-    //     alertName: alertName,
-    //   });
-    // } else {
-    //   instantSubscribe({
-    //     alertConfiguration: null,
-    //     alertName: alertName,
-    //   });
-    // }
-  };
+    setErrorMessage('');
+    if (!enabled && initialRatio) {
```

Comment:
i think we should do an explicit check on `initialRatio !== null` because a user might want to set an alert at 0% health